### PR TITLE
XRI3 migration xri2 to xri3 migration guide

### PR DESCRIPTION
**Pull request by ms-RistoRK**
* Updated README.md with a section regarding XRI2 -> XRI3 migration
* Added XRI3TOXRI2MIGRATIONGUIDE.md as a guide for XRI3 migration.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,6 +80,26 @@ This project welcomes contributions, suggestions, and feedback. All contribution
...
 For more information on how to contribute Mixed Reality Toolkit for Unity Project, please read [CONTRIBUTING.md](./CONTRIBUTING.md).
 
+## MRTK3 XRI2 to XRI3 migration guide
+
+MRTK3 has been upgraded to use [Unity's XR Interaction Toolkit 3+](https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@3.0/manual/whats-new-3.0.html).  As part of the upgrade several changes were made to properly consume XRI 3 package and adhere to the new patterns.  In a nutshell, the main changes are summarized as follows:
+
+* New controller prefabs and a new rig have been created following the new XRI3 pattern.
+  * The old controllers and rig have been marked as osbolete and renamed as "Obsolete MRTK XR Rig", "Obsolete MRTK LeftHand Controller", "Obsolete MRTK RightHand Controller", "Obsolete MRTK Hand Controller", "Obsolete MRTK Interaction Manager", and "Obsolete MRTK Gaze Controller".
+  * The new controllers and rig retake the original names of the obsolete controllers.
+* New controllers structure have been uniformized so that all of them have the same structure.
```
Comment (shaynie) [inline README.md:90]: have been modified so that all of them have the same structure.

File: XRI3TOXRI2MIGRATIONGUIDE.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# MRTK3 XRI2 to XRI3 migration guide
+
+## Introduction
+
+Thank you for choosing MRTK as your Mixed Reality framework.  MRTK has been upgraded to follow XRI3 patterns and consume [Unity's XR Interaction Toolkit 3+](https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@3.0/manual/whats-new-3.0.html) package.  The upgrade encompass various changes from the previous MRTK version and we want to share our experience during the upgrade so that the community is aware of the changes that had to be implemented.  We also want to provide an overall roadmap for others to do their own upgrades in their forks or propietary solutions and therefore we created this guide.  We created this guide to empower you and your organization to achieve more in Mixed Reality solutions.
```
Comment (shaynie) [inline XRI3TOXRI2MIGRATIONGUIDE.md:5]: encompasses

File: XRI3TOXRI2MIGRATIONGUIDE.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+* Adhere to the new XRI3 patterns.
+* Provide backward compatibility.
+
+As you will notice in the code many of our changes query the Input Compatibility Mode field to differentiate between XRI2 functionality and XRI3 functionality.  This is not mandatory for a successful XRI3 migration, however, we wanted to create a smooth transition for all of you and did our best to ensure backward compatibily as you migrate your forks or solutions.
+
+We encourage you to check the code in each script as well as the commit history to get a sense of the changes that were needed.
+
+## Step 5 - Update Unity-tests
+
+Last but not least, Unity-tests!.  We created several new Unity-tests to ensure the correct functionality of the new XRI3 MRTK rig as well as the old functionality.  We encourage you to visit the code and commit history related to Unity-tests to get a sense of the changes.  Note that the changes focus mostly on where and how the input actions are queried but the overal functionality test remains the same.
```
Comment (shaynie) [inline XRI3TOXRI2MIGRATIONGUIDE.md:84]: One punctuation mark, please :)

**Review by shaynie — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ms-RistoRK) [reply README.md]: Thx for feedback, implemented in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/839/commits/e14267bbcb572fa95f69535cbc5fc9cbb0a22d00, resolving.

File: XRI3TOXRI2MIGRATIONGUIDE.md (hunk 2 above)
Comment (ms-RistoRK) [reply XRI3TOXRI2MIGRATIONGUIDE.md]: Thx for feedback, implemented in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/839/commits/e14267bbcb572fa95f69535cbc5fc9cbb0a22d00, resolving.

File: XRI3TOXRI2MIGRATIONGUIDE.md (hunk 3 above)
Comment (ms-RistoRK) [reply XRI3TOXRI2MIGRATIONGUIDE.md]: Thx for feedback, implemented in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/839/commits/e14267bbcb572fa95f69535cbc5fc9cbb0a22d00, resolving.

**Review by shaynie — APPROVED** [review]

File: XRI3TOXRI2MIGRATIONGUIDE.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+## Introduction
+
+Thank you for choosing MRTK as your Mixed Reality framework.  MRTK has been upgraded to follow XRI3 patterns and consume [Unity's XR Interaction Toolkit 3+](https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@3.0/manual/whats-new-3.0.html) package.  The upgrade encompasses various changes from the previous MRTK version and we want to share our experience during the upgrade so that the community is aware of the changes that had to be implemented.  We also want to provide an overall roadmap for others to do their own upgrades in their forks or propietary solutions and therefore we created this guide.  We created this guide to empower you and your organization to achieve more in Mixed Reality solutions.
+
+## Original MRTK rig structure
+
+Previous to XRI3 upgrade the MRTK3 rig had a structure similar to the one shown in the next figure:
+
+<img width="522" alt="PreXRI3Structure" src="https://github.com/ms-RistoRK/MixedRealityToolkit-Unity/assets/84108471/1b104289-83a1-4b90-8d30-61bb03eae635">
```
Comment (whebertML) [inline XRI3TOXRI2MIGRATIONGUIDE.md:11]: Won't it be an issue that all these image links are links to a personal github account, not to the main mixed reality group's account?

Comment (ms-RistoRK) [reply XRI3TOXRI2MIGRATIONGUIDE.md]: Images references fixed in 26e2c268ea30496a8a6da82a5ddca61bab0f2597 , resolving.

**Review by shaynie — APPROVED** [review]

**Pull request merged**